#! /bin/sh /usr/share/dpatch/dpatch-run
## comment_support.dpatch by Proformatique
##
## DP: add support for "commented" lines in tables

@DPATCH@
--- a/res/res_config_mysql.c	2008-12-11 17:55:16.000000000 +0100
+++ b/res/res_config_mysql.c	2008-12-11 17:56:54.000000000 +0100
@@ -69,6 +69,7 @@
 static char   dbname[50];
 static char   dbsock[50];
 static int    dbport;
+static int    comment_support;
 static int    connected;
 static time_t connect_time;
 
@@ -126,7 +127,12 @@
 	if ((valsz = strlen (newval)) * 2 + 1 > sizeof(buf))
 		valsz = (sizeof(buf) - 1) / 2;
 	mysql_real_escape_string(&mysql, buf, newval, valsz);
-	snprintf(sql, sizeof(sql), "SELECT * FROM %s WHERE %s%s '%s'", table, newparam, op, buf);
+
+	if (!comment_support)
+		snprintf(sql, sizeof(sql), "SELECT * FROM %s WHERE %s%s '%s'", table, newparam, op, buf);
+	else
+		snprintf(sql, sizeof(sql), "SELECT * FROM %s WHERE NOT commented AND %s%s '%s'", table, newparam, op, buf);
+
 	while((newparam = va_arg(ap, const char *))) {
 		newval = va_arg(ap, const char *);
 		if(!strchr(newparam, ' ')) op = " ="; else op = "";
@@ -241,7 +247,12 @@
 	if ((valsz = strlen (newval)) * 2 + 1 > sizeof(buf))
 		valsz = (sizeof(buf) - 1) / 2;
 	mysql_real_escape_string(&mysql, buf, newval, valsz);
-	snprintf(sql, sizeof(sql), "SELECT * FROM %s WHERE %s%s '%s'", table, newparam, op, buf);
+
+	if (!comment_support)
+		snprintf(sql, sizeof(sql), "SELECT * FROM %s WHERE %s%s '%s'", table, newparam, op, buf);
+	else
+		snprintf(sql, sizeof(sql), "SELECT * FROM %s WHERE NOT commented AND %s%s '%s'", table, newparam, op, buf);
+
 	while((newparam = va_arg(ap, const char *))) {
 		newval = va_arg(ap, const char *);
 		if(!strchr(newparam, ' ')) op = " ="; else op = "";
@@ -579,6 +590,12 @@
                 } else {
                         strncpy(dbsock, s, sizeof(dbsock) - 1);
                 }
+
+		if(!(s=ast_variable_retrieve(config, "general", "comment_support"))) {
+			comment_support = 0;
+		} else {
+			comment_support = ast_true(s);
+		}
 	}
 	ast_config_destroy(config);
 
